### PR TITLE
Fix experimental `remotePatterns` wildcard

### PR DESCRIPTION
**Pull request by styfle**
- Follow up to #36245 
- Closes #37026

### CODE REVIEW

File: packages/next/build/index.ts [hunk 1]
```diff
@@ -2216,6 +2218,15 @@ export default async function build(
       ;(images as any).sizes = [...deviceSizes, ...imageSizes]
       ;(images as any).remotePatterns =
```
Comment (styfle) [inline packages/next/build/index.ts:2219]: @ijjk Do we care about being backwards compatible here? Its experimental so I'm wonder if we want to keep this as-is or just break it now to always return regex?

**Comment by ijjk** [comment]
## Failing test suites 
Commit: 96598366802bfbd739051333a5b29c8f0534d1c1

`yarn testheadless test/e2e/browserslist-extends/index.test.ts` 
- browserslist-extends > should work

<details>
<summary>Expand output</summary>

  ● browserslist-extends › should work

    FetchError: request to http://localhost:45153/ failed, reason: socket hang up

      at ClientRequest.<anonymous> (../node_modules/node-fetch/lib/index.js:1491:11)

</details>


Read more about building and testing Next.js in [contributing.md](https://github.com/vercel/next.js/blob/canary/contributing.md).

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Build</strong> (Decrease detected ✓)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall decrease ✓</summary>

|  | vercel/next.js canary  | vercel/next.js use-micromatch-regex-remote-patterns | Change |
| - | - | - | - |
| buildDuration | 15.9s | 15.7s | -199ms |
| buildDurationCached | 6.5s | 6.4s | -92ms |
| nodeModulesSize | 479 MB | 479 MB | -1.71 kB |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall decrease ⚠️</summary>

|  | vercel/next.js canary  | vercel/next.js use-micromatch-regex-remote-patterns | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 3.752 | 3.762 | ⚠️ +0.01 |
| / avg req/sec | 666.34 | 664.58 | ⚠️ -1.76 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.166 | 1.179 | ⚠️ +0.01 |
| /error-in-render avg req/sec | 2143.21 | 2120.08 | ⚠️ -23.13 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack)</strong></summary>

|  | vercel/next.js canary  | vercel/next.js use-micromatch-regex-remote-patterns | Change |
| - | - | - | - |
| 925.HASH.js gzip | 179 B | 179 B | ✓ |
| framework-HASH.js gzip | 42 kB | 42 kB | ✓ |
| main-HASH.js gzip | 29.2 kB | 29.2 kB | ✓ |
| webpack-HASH.js gzip | 1.54 kB | 1.54 kB | ✓ |
| Overall change | 73 kB | 73 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | vercel/next.js use-micromatch-regex-remote-patterns | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | vercel/next.js use-micromatch-regex-remote-patterns | Change |
| - | - | - | - |
| _app-HASH.js gzip | 1.36 kB | 1.36 kB | ✓ |
| _error-HASH.js gzip | 193 B | 193 B | ✓ |
| amp-HASH.js gzip | 308 B | 308 B | ✓ |
| css-HASH.js gzip | 327 B | 327 B | ✓ |
| dynamic-HASH.js gzip | 2.71 kB | 2.71 kB | ✓ |
| head-HASH.js gzip | 359 B | 359 B | ✓ |
| hooks-HASH.js gzip | 920 B | 920 B | ✓ |
| image-HASH.js gzip | 5.73 kB | 5.73 kB | ✓ |
| index-HASH.js gzip | 263 B | 263 B | ✓ |
| link-HASH.js gzip | 2.65 kB | 2.65 kB | ✓ |
| routerDirect..HASH.js gzip | 320 B | 320 B | ✓ |
| script-HASH.js gzip | 391 B | 391 B | ✓ |
| withRouter-HASH.js gzip | 318 B | 318 B | ✓ |
| 85e02e95b279..7e3.css gzip | 107 B | 107 B | ✓ |
| Overall change | 16 kB | 16 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | vercel/next.js use-micromatch-regex-remote-patterns | Change |
| - | - | - | - |
| _buildManifest.js gzip | 458 B | 458 B | ✓ |
| Overall change | 458 B | 458 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | vercel/next.js use-micromatch-regex-remote-patterns | Change |
| - | - | - | - |
| index.html gzip | 533 B | 533 B | ✓ |
| link.html gzip | 547 B | 547 B | ✓ |
| withRouter.html gzip | 529 B | 529 B | ✓ |
| Overall change | 1.61 kB | 1.61 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Default Build with SWC</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall decrease ✓</summary>

|  | vercel/next.js canary  | vercel/next.js use-micromatch-regex-remote-patterns | Change |
| - | - | - | - |
| buildDuration | 17.9s | 17.8s | -118ms |
| buildDurationCached | 6.4s | 6.4s | ⚠️ +12ms |
| nodeModulesSize | 479 MB | 479 MB | -1.71 kB |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | vercel/next.js canary  | vercel/next.js use-micromatch-regex-remote-patterns | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 3.79 | 3.794 | 0 |
| / avg req/sec | 659.66 | 659.01 | ⚠️ -0.65 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.184 | 1.169 | -0.01 |
| /error-in-render avg req/sec | 2111.08 | 2138.51 | +27.43 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack)</strong></summary>

|  | vercel/next.js canary  | vercel/next.js use-micromatch-regex-remote-patterns | Change |
| - | - | - | - |
| 925.HASH.js gzip | 178 B | 178 B | ✓ |
| framework-HASH.js gzip | 42.7 kB | 42.7 kB | ✓ |
| main-HASH.js gzip | 29.7 kB | 29.7 kB | ✓ |
| webpack-HASH.js gzip | 1.54 kB | 1.54 kB | ✓ |
| Overall change | 74.1 kB | 74.1 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | vercel/next.js use-micromatch-regex-remote-patterns | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | vercel/next.js use-micromatch-regex-remote-patterns | Change |
| - | - | - | - |
| _app-HASH.js gzip | 1.35 kB | 1.35 kB | ✓ |
| _error-HASH.js gzip | 179 B | 179 B | ✓ |
| amp-HASH.js gzip | 311 B | 311 B | ✓ |
| css-HASH.js gzip | 324 B | 324 B | ✓ |
| dynamic-HASH.js gzip | 2.89 kB | 2.89 kB | ✓ |
| head-HASH.js gzip | 357 B | 357 B | ✓ |
| hooks-HASH.js gzip | 920 B | 920 B | ✓ |
| image-HASH.js gzip | 5.82 kB | 5.82 kB | ✓ |
| index-HASH.js gzip | 261 B | 261 B | ✓ |
| link-HASH.js gzip | 2.78 kB | 2.78 kB | ✓ |
| routerDirect..HASH.js gzip | 322 B | 322 B | ✓ |
| script-HASH.js gzip | 392 B | 392 B | ✓ |
| withRouter-HASH.js gzip | 317 B | 317 B | ✓ |
| 85e02e95b279..7e3.css gzip | 107 B | 107 B | ✓ |
| Overall change | 16.3 kB | 16.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | vercel/next.js use-micromatch-regex-remote-patterns | Change |
| - | - | - | - |
| _buildManifest.js gzip | 457 B | 457 B | ✓ |
| Overall change | 457 B | 457 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | vercel/next.js use-micromatch-regex-remote-patterns | Change |
| - | - | - | - |
| index.html gzip | 531 B | 531 B | ✓ |
| link.html gzip | 545 B | 545 B | ✓ |
| withRouter.html gzip | 527 B | 527 B | ✓ |
| Overall change | 1.6 kB | 1.6 kB | ✓ |

</details>

</details>
Commit: 96598366802bfbd739051333a5b29c8f0534d1c1

File: packages/next/build/index.ts (hunk 1 above)
Comment (styfle) [reply packages/next/build/index.ts]: I think we can tell the difference between old and new format by looking for the leading `^`

**Review by ijjk — DISMISSED** [review]

File: packages/next/build/index.ts (hunk 1 above)
Comment (ijjk) [reply packages/next/build/index.ts]: I think checking for `^` should be good, although it should also be ok to make breaking change here if needed

**Review by ijjk — APPROVED** [review]

**Pull request merged**